Content.Server/Chemistry/Components/SolutionTransferComponent.cs

```diff
@@ -129,19 +129,21 @@ async Task<bool> IAfterInteract.AfterInteract(AfterInteractEventArgs eventArgs)
                 return false;
             }
 
-
-            if (CanReceive && _entities.TryGetComponent(target, out ReagentTankComponent? tank)
-                           && solutionsSys.TryGetRefillableSolution(Owner, out var ownerRefill)
-                           && solutionsSys.TryGetDrainableSolution(target, out var targetDrain))
+            //Special case for reagent tanks, because normally clicking another container will give solution, not take it.
+            if (CanReceive  && _entities.TryGetComponent(target, out DrainableSolutionComponent? drainable) // target must be drainable
+                            && !_entities.TryGetComponent(target, out RefillableSolutionComponent? refillable) // target must not be refillable (e.g. Reagent Tanks)
+                            && solutionsSys.TryGetRefillableSolution(Owner, out var ownerRefill)
+                            && solutionsSys.TryGetDrainableSolution(target, out var targetDrain))
```

Comment:
```suggestion
            if (CanReceive  && !_entities.HasComponent<RefillableSolutionComponent>(target) // target must not be refillable (e.g. Reagent Tanks)
                            && _entities.TryGetComponent(Owner, out RefillableSolutionComponent ownerRefull)
                            && solutionsSys.TryGetRefillableSolution(Owner, out var ownerRefill, refillable: ownerRefull)
                            && solutionsSys.TryGetDrainableSolution(target, out var targetDrain)) // target must be drainable
```
Re-use `ownerRefill` instead of doing another `TryGet` later on, and just remove the `HasComponent<SolutionContainerManagerComponent>` checks above

Reply:
Ah perfect! I understand now, thank you. Was about to comment that the `HasComponent<SolutionContainerManagerComponent>` checks may be needed elsewhere because they weren't mine, but then I saw your earlier comment addressing that.